baseapp/baseapp.go

```diff
@@ -233,6 +234,30 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 	app.setDeliverState(abci.Header{})
 	app.initChainer(app.deliverState.ctx, req) // no error
 
+	// Initialize module genesis state
+	genesisState := new(map[string]json.RawMessage)
+	err := json.Unmarshal(req.AppStateBytes, genesisState)
+	if err != nil {
+		// TODO Return something intelligent
+		panic(err)
+	}
+	err = app.Router().ForEach(func(r string, _ sdk.Handler, i sdk.InitGenesis) error {
```

Comment:
should reference a type which is that `func(r string, _ sdk.Handler, i sdk.InitGenesis) error`